clients/src/test/java/org/apache/kafka/common/requests/ProduceRequestTest.java

```diff
@@ -120,15 +121,38 @@ public void testBuildWithCurrentMessageFormat() {
         ProduceRequest.Builder requestBuilder = ProduceRequest.forMagic(RecordBatch.CURRENT_MAGIC_VALUE,
                 new ProduceRequestData()
                         .setTopicData(new ProduceRequestData.TopicProduceDataCollection(Collections.singletonList(
-                                new ProduceRequestData.TopicProduceData().setName("test").setPartitionData(Collections.singletonList(
-                                        new ProduceRequestData.PartitionProduceData().setIndex(9).setRecords(builder.build()))))
+                                        new ProduceRequestData.TopicProduceData()
+                                                .setName("test")
```

Comment:
will we have a case we write both?

Reply:
No we don't have a case where we will write both. However if we passed both the request will drop the name with version >= 12. I'll remove `setName` from this test so we don't give the wrong impression that we are expecting both